interpreter/voice.py

```diff
@@ -0,0 +1,56 @@
+import time
+import threading
+from mutagen.mp3 import MP3
+from gtts import gTTS
+import speech_recognition as SR
+import pygame
+import os
+os.environ['PYGAME_HIDE_SUPPORT_PROMPT'] = "hide"
+
+
+class SpeechAssistant:
+    def __init__(self, wakeup_word="interpreter"):
+        self.wakeup_word = wakeup_word
+        self.speech_recognizer = SR.Recognizer()
+
+    def tts_and_play_audio(self, text):
+        language = 'en'
+        speech = gTTS(text=text, lang=language, slow=False, tld='com.au')
+        speech.save("speech.mp3")
+        self.play_audio("speech.mp3")
+        os.remove("speech.mp3")
+
+    def play_audio(self, file_path):
+        pygame.mixer.init()
+        pygame.mixer.music.load(file_path)
+        pygame.mixer.music.play()
+        audio = MP3(file_path)
+        time.sleep(audio.info.length)
+        pygame.mixer.music.stop()
+        pygame.mixer.quit()
+        # Delete the file after playing it
+
+    def start_speech_recognition(self):
+        while True:
+             print("Listening for wakeup word...")
+             with SR.Microphone() as mic:
+                   try:
+                        self.speech_recognizer.adjust_for_ambient_noise(
+                            mic, duration=0.4)
+                        audio = self.speech_recognizer.listen(mic)
+                        text = self.speech_recognizer.recognize_google(
```

Comment:
This seems to depend on online service (similar to text to speech above?). That means it leaks information even if open-interpreter is used with local LLM.

Reply:
There is local TTS Services as well but changes from system to system so we should use that instead

Reply:
What do we think of pyttsx3 looks like a good alternative to gtts can run locally and should work well on windows, Mac and linux?

Reply:
> What do we think of pyttsx3 looks like a good alternative to gtts can run locally and should work well on windows, Mac and linux?

I have used on mac we need to have a wrapper library which check which OS and which package to use from python because sometimes it doesnt speak no voice available i had issue with my previous projects.

Reply:
I have seen this [GPT4ALL Voice for GPT](https://github.com/Ai-Austin/GPT4ALL-Voice-Assistant) project thought it would be useful here.

Reply:
Yeah that project is using pyttsx3 and speech recognition with SpeechRecognition and whisper which I've only now just got running on my system 😅 but yeah probably the way to go ☺️ it's seems much faster at loading the base model also compared to everything else I've tried

Reply:
Or if you find any other project which is faster and reliable than this you can use it. 
Good luck implementing this into open-interpreter.

Reply:
much simpler tts with pytts and speech recognition with whisper as prommised still need to add code to check os or highjack the already existing fuctions